test/Worker/ProcessPoolErrorTest.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Amp\Parallel\Test\Worker;
+
+use Amp\Parallel\Worker\DefaultPool;
+use Amp\Parallel\Worker\Pool;
+use Amp\Parallel\Worker\WorkerFactory;
+use Amp\Parallel\Worker\WorkerProcess;
+
+/**
+ * @group process
+ */
+class ProcessPoolErrorTest extends AbstractPoolErrorTest {
+    protected function createPool($max = Pool::DEFAULT_MAX_SIZE): Pool {
+        $factory = $this->createMock(WorkerFactory::class);
+        $factory->method('create')->will($this->returnCallback(function () {
+            return new WorkerProcess;
+        }));
+
+        return new DefaultPool(-1, $factory);
```

Comment:
`$max` is completely ignored here.

Reply:
To tests the ```Exception```, I will modify the ```$max``` value.

Reply:
Shouldn't the test then just call `createPool(-1)`?